app/Models/Exchange.php

```diff
@@ -4,10 +4,33 @@
 
 namespace App\Models;
 
+use Illuminate\Database\Eloquent\Builder;
 use Illuminate\Database\Eloquent\Factories\HasFactory;
 use Illuminate\Database\Eloquent\Model;
 
 final class Exchange extends Model
 {
     use HasFactory;
+
+    public function scopeFilterByType(Builder $query, ?string $type): Builder
+    {
+        if ($type === 'exchanges') {
+            return $query->where('is_exchange', true);
+        }
+
+        if ($type === 'aggregators') {
+            return $query->where('is_aggregator', true);
+        }
+
+        return $query;
```

Comment:
Alternative approach if you're interested:

```suggestion
        return $query
            ->when($type === 'exchanges', static fn ($query) => $query->where('is_exchange', true))
            ->when($type === 'aggregators', static fn ($query) => $query->where('is_aggregator', true));
```